src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php

```diff
@@ -82,8 +114,13 @@ public function getTypeGuesser()
             $this->guesserLoaded = true;
             $guessers = array();
 
-            foreach ($this->guesserServiceIds as $serviceId) {
-                $guessers[] = $this->container->get($serviceId);
+            foreach ($this->guesserServices as $serviceId => $service) {
+                if (is_string($service)) {
+                    $serviceId = $service;
+                    $service = $this->typeContainer->get($serviceId);
+                }
+
+                $guessers[] = $service;
```

Comment:
Not sure we need `FormTypeGuesserChain` to be a service, but simply pass it the guessers iterator (so allow `FormTypeGuesserChain` to accept iterable).
IMHO the extension is in charge of building the whole thing and the use of the `FormTypeGuesserChain` is an implementation detail. There is no need to pass the guesser chain as a dependency.

There is an issue with `FormTypeGuesserChain`, though... its constructor. It iterates over the guessers and this would trigger the loading too soon anyway.

I'm not sure lazy loading here is a big deal though, as `FormExtensionInterface::getTypeGuesser()` is called in `FormRegistryInterface::getTypeGuesser()`, called in `Formfactory::createBuilderForProperty()` and is used directly to guess types by iterating over every guessers.